### PR TITLE
Fix block uncompress error in SquashfsReadFile

### PR DESCRIPTION
**Pull request by wcampbell0x2a**

### CODE REVIEW

**Comment by wcampbell0x2a** [comment]
@rbran Found this error while testing backhand's `unsquashfs`.

Looks like there is still an error with the blocks somewhere? I see a bunch of file differs from running the following commands. Any ideas?
```
$ ./target/release/unsquashfs -f -d out-rust test-assets/test_appimage_firefox/bytes.squashfs
$ unsquashfs -d out-c test-assets/test_appimage_firefox/bytes.squashfs
$ diff -bur out-c out-rust
```

For example, it's like every file is off by a block or something?
```
$ cat out-rust/gmp-clearkey/0.1/manifest.json
[Desktop Entry]
Version=1.0
Name=Firefox
GenericName=Web Browser
GenericName[ar]=متصفح ويب
GenericName[ast]=Restolador Web
GenericName[bn]=ওয়েব ব্রাউজার
GenericName[ca]=Navegador web
Generic⏎                                                                                                                                                                                                                                                                                                                                 

$ cat out-c/gmp-clearkey/0.1/manifest.json
{
    "name": "clearkey",
    "description": "ClearKey Gecko Media Plugin",
    "version": "1",
    "x-cdm-module-versions": "4",
    "x-cdm-interface-versions": "10",
    "x-cdm-host-versions": "10",
    "x-cdm-codecs": ""
}⏎
```

File: src/filesystem.rs [hunk 1]
```diff
@@ -319,11 +320,13 @@ impl<'a, R: ReadSeek> SquashfsReadFile<'a, R> {
             None => return Ok(()),
         };
         if block.uncompressed {
-            self.raw_data.decompress(&block, &mut self.buf_decompress)?;
-        } else {
             //data is already decompress, so just swap the read and decompress
             //buffers, so the buf_decompress contains the final data.
             std::mem::swap(&mut self.buf_read, &mut self.buf_decompress);
+            self.buf_decompress =
```
Comment (wcampbell0x2a) [inline src/filesystem.rs:326]: @rbran  Found the problem! We need to use `block_offset` correctly to read the right start position.

Any way of getting rid of the `.to_vec()` tho ?  :(

Comment (rbran) [reply src/filesystem.rs:326]: Is possible to make this operation directly on the Vec using drain, like in https://github.com/wcampbell0x2a/backhand/pull/147

**Comment by rbran** [comment]
Should the offset be applied to both compress/uncompressed if is a fragment?

Also I had a problem with fragments bound only by the start, so I created fragment_range.

In any case I'm creating a patch to solve all those things.

**Pull request merged**